org.eclipse.scanning.points/src/org/eclipse/scanning/points/PointGeneratorService.java

```diff
@@ -158,13 +158,13 @@
 		}
 	}
 
-	private static void fillStaticGeneratorInfo(Map<Class<? extends IScanPathModel>, Class<? extends IPointGenerator>> gens, Map<String,   GeneratorInfo> ids) {
+	private static void fillStaticGeneratorInfo(Map<Class<? extends IScanPathModel>, Class<? extends IPointGenerator<?>>> gens, Map<String,   GeneratorInfo> ids) {
 
-		for (Class<? extends IScanPathModel> modelClass : gens.keySet()) {
+		for (Map.Entry<Class<? extends IScanPathModel>, Class<? extends IPointGenerator<?>>> genEntry : gens.entrySet()) {
 			try {
 				final GeneratorInfo info = new GeneratorInfo();
-				info.setModelClass(modelClass);
-				info.setGeneratorClass(gens.get(modelClass));
+				info.setModelClass(genEntry.getKey());
+				info.setGeneratorClass(genEntry.getValue());
 				ids.put(info.getGeneratorClass().newInstance().getId(), info);
 			} catch (Exception e) {
 				e.printStackTrace();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1148)